src/Controller/ShopwareFilesController.php

```diff
@@ -0,0 +1,188 @@
+
+            if (!is_file($path)) {
+                continue;
+            }
+
+            $md5Sum = md5_file($path);
+
+            $ignoredState = $this->isIgnoredFileHash($file, $md5Sum);
+            if ($ignoredState === self::STATUS_IGNORED_ALL) {
+                continue;
+            }
+
+            if ($md5Sum !== $expectedMd5Sum) {
+                if ($ignoredState === self::STATUS_OK) {
+                    $allFilesAreOkay = false;
+                }
+
+                $invalidFiles[] = [
+                    'name' => $file,
+                    'shopwareUrl' => $this->getShopwareUrl($file),
+                    'expected' => $ignoredState === self::STATUS_IGNORED_IN_PROJECT,
+                ];
+            }
+        }
+
+        if ($allFilesAreOkay) {
+            return new JsonResponse(['ok' => true, 'files' => $invalidFiles]);
+        }
+
+        return new JsonResponse(['ok' => false, 'files' => $invalidFiles]);
+    }
+
+    #[Route(path: '/file-contents', name: 'api.frosh.tools.file-contents', methods: ['GET'])]
+    public function getFileContents(Request $request): JsonResponse
+    {
+        if ($this->shopwareVersion === Kernel::SHOPWARE_FALLBACK_VERSION) {
+            return new JsonResponse(['error' => 'Git version is not supported']);
+        }
+
+        $file = $request->query->get('file');
+        if (!$file) {
+            return new JsonResponse(['error' => 'no file provided']);
+        }
+
+        $path = realpath($this->projectDir . '/' . $file);
+        if ($path === false || !str_starts_with($path, $this->projectDir) || !is_file($path)) {
+            return new JsonResponse(['error' => 'File is invalid']);
+        }
+
+        return new JsonResponse([
+            'name' => $file,
+            'shopwareUrl' => $this->getShopwareUrl($file),
+            'content' => file_get_contents($path),
+            'originalContent' => $this->getOriginalFileContent($file),
+        ]);
+    }
+
+    #[Route(path: '/shopware-file/restore', name: 'api.frosh.tools.shopware-file.restore', methods: ['GET'])]
+    public function restoreShopwareFile(Request $request): JsonResponse
+    {
+        if ($this->shopwareVersion === Kernel::SHOPWARE_FALLBACK_VERSION) {
+            return new JsonResponse(['error' => 'Git version is not supported']);
+        }
+
+        $file = $request->query->get('file');
+        if (!$file) {
+            return new JsonResponse(['error' => 'no file provided']);
+        }
+
+        $path = realpath($this->projectDir . '/' . $file);
+        if ($path === false || !str_starts_with($path, $this->projectDir) || !is_file($path)) {
+            return new JsonResponse(['error' => 'File is invalid']);
+        }
+
+        $content = $this->getOriginalFileContent($file);
+        if ($content === null || $content === '') {
+            return new JsonResponse(['error' => 'File would be empty!']);
+        }
+
+        file_put_contents($path, $content);
```

Comment:
You need to call `opcache_reset()` with an if condition around with function_exists